cscs-checks/libraries/petsc/petsc_helloworld.py

```diff
@@ -28,21 +19,18 @@ def __init__(self, variant, **kwargs):
         self.num_tasks = 16
         self.num_tasks_per_node = 8
         self.dynamic = True if variant == 'dynamic' else False
-        self.executable_opts = ['-da_grid_x 4', '-da_grid_y 4', '-mat_view',
-                                '> petsc_poisson2d.out']
+        self.executable_opts = ['-da_grid_x 4', '-da_grid_y 4', '-ksp_monitor']
 
-        self.maintainers = ['WS', 'AJ']
-        self.tags = {'production'}
+        norms = sn.extractall(r'\s+\d+\s+KSP Residual norm\s+(?P<norm>\S+)',
+                              self.stdout, 'norm', float)
 
-        self.sanity_patterns = sanity_filecmp(
-            'petsc_poisson2d.out', 'petsc_poisson2d.ref')
+        # Check the final residual norm for convergence
+        self.sanity_patterns = sn.assert_lt(norms[-1], 1.0e-5)
```

Comment:
Since you don't do anything with the rest of the norms that you extract, you could use `sn.extractsingle()` and pass `item=-1`, so to extract the last match. The advantage of this is that it will raise a `SanityError` instead of `IndexError` in case of an out-of-bounds error, which means that it will be printed nicely in the failure report. See [here](https://eth-cscs.github.io/reframe/sanity_functions_reference.html#reframe.utility.sanity.extractsingle) for the full spec of the `extractsingle()` function.